src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/MetricsCollectorToolSpark.java

```diff
@@ -0,0 +1,94 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines.metrics;
+
+import htsjdk.samtools.SAMFileHeader;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.metrics.MetricsArgs;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Base class for standalone Spark metrics collector tools.ß
+ *
+ */
+public abstract class MetricsCollectorToolSpark<T extends MetricsArgs>
```

Comment:
The reason for having this tool (which is really a base class that defines a special kind of a tool) declare that it implements this interface is not to allow them to be called from the outside; its that the interface is what defines the requirements of a subclass. If you look at the runTool method below, it needs to be able to call the implementations of those interface methods in order to put the collector through it's lifecycle. I don't think its any different than saying, for example, that a ReadWalker has to implement an "apply" method; nobody outside of the tool calls it; but the base ReadWalker tool depends on subclasses implementing it (and in this case the interface is used in several other places). If I remove it, I have to implement another layer of delegation that I think would add complexity but no real value. Having said that, let me know if you still think its the right thing to do.